src/realm/group_shared.cpp

```diff
@@ -1709,10 +1719,39 @@ void SharedGroup::do_end_read() noexcept
 void SharedGroup::do_begin_write()
 {
     SharedInfo* info = m_file_map.get_addr();
-    // Get write lock
-    // Note that this will not get released until we call
-    // commit() or rollback()
+    // Get write lock - the write lock is held until do_end_write().
+    // 
+    // We use a ticketing scheme to ensure fairness wrt performing write transactions.
+    // (But cannot do that on Windows until we have interprocess condition variables there)
+#ifndef _WIN32
+    uint_fast32_t my_ticket = info->next_ticket.fetch_add(1, std::memory_order_relaxed);
+#endif
     m_writemutex.lock(); // Throws
+#ifndef _WIN32
+    // allow for comparison even after wrap around of ticket numbering:
+    int32_t diff = int32_t(my_ticket - info->next_served);
+    bool should_yield = diff > 0; // ticket is in the future
```

Comment:
Where did the while loop go? It seems like the first one woken up will always proceed. So where does the fairness come in?